### ISSUE TITLE
Dragging the page leads to high frequent jumps

### ISSUE THREAD

**Issue by funnym0nk3y** [xournalpp/xournalpp#3260]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Win10
 - Which version of libgtk do you use:
     3.24.30
 - Version of Xournal++:
     9d1277de
 - Installation method: exe

**Describe the bug**
Dragging the page with a touchscreen device leads to high frequent high amplitude jumps of the page. Touching it once with a second finger solves it.

**Comment by Technius** [comment]
Does this occur if you disable zoom fit?

**Comment by personalizedrefrigerator** [comment]
#2837 might fix this (but I'm not sure!)...

**Comment by funnym0nk3y** [comment]
@Technius It is independent of the zoom level.

**Comment by personalizedrefrigerator** [comment]
I can reproduce this with the flatpak installation on Fedora 34/KDE Plasma/Wayland.

**Comment by lb90** [comment]
@funnym0nk3y does it happen when scrolling in other windows, e.g. the preferences dialog?

**Comment by funnym0nk3y** [comment]
@lb90 I tried it in the font drop-down where everything was fine. I can't drag in the preferences dialog though.

**Comment by lb90** [comment]
> I can't drag in the preferences dialog though.

Can you test with gtk3-demo from the automated CI builds? [Download](https://gitlab.gnome.org/lb90/gtk/-/jobs/1417729/artifacts/file/_build/demos/gtk-demo/gtk3-demo.exe) the file and copy it inside `Xournal++\bin`. The left pane scrolls with touch input:

| gtk3-demo |
| --- |
| ![gtk3-demo](https://user-images.githubusercontent.com/12200013/127741445-f245aa77-14ce-4a30-8412-b2e41f0945df.png) |

**Comment by lb90** [comment]
It's also worth trying two tests in the list:

* Gestures
* Touch and Drawing Tablets

**Comment by funnym0nk3y** [comment]
I tired all three tests. Scrolling works fine, the zoom gesture also and using the pen everything is fine. But the dragging gesture seems to function not properly. It is not jumping with high frequency but the results are very inconsistent and do not resemble what was done on the screen.

**Comment by lb90** [comment]
If you feel like, could you make a short screen recording?

**Comment by funnym0nk3y** [comment]
Yeah, I'm still thinking about a way so that you can see my touch input.

**Comment by lb90** [comment]
Maybe you can try the built-in screen recorder of Windows 10: https://betanews.com/2019/01/14/windows-10-screen-recorder-ultility/ (not sure if it shows touch input, but it's worth a try)

**Comment by personalizedrefrigerator** [comment]
> I tired all three tests. Scrolling works fine, the zoom gesture also and using the pen everything is fine. But the dragging gesture seems to function not properly. It is not jumping with high frequency but the results are very inconsistent and do not resemble what was done on the screen.

#3379 might work around this bug. Inertial scrolling won't work, though.

**Comment by PiiNGPooNG** [comment]
I think I'm experiencing this issue as well. I'm using a Surface Go 2 with Windows 10. Xournal++ 1.1.0 (Hotfix 1)

https://youtu.be/WgYs5DAgZec

At the beginning I'm trying to slowly scroll down. Once it jumped down to the last page (0:09) I start flicking up. Once I reach the top of the document (00:15) I start flicking down.

The jumping occurs while my finger is touching the screen. So when slowly scrolling it jumps around a lot. Flicking can work but is not reliable either.

**Comment by lb90** [comment]
Hello @PiiNGPooNG, thanks for the feedback! First, I'd need to know if it's an issue in Xournal++ or in GTK3. Can you try with the GTK3 Demo and report wether it works? See https://github.com/xournalpp/xournalpp/issues/3260#issuecomment-890350546

**Comment by lb90** [comment]
You can also test three specific demos:

* Gestures
* Paint
* Touch and drawing tablets

**Comment by lb90** [comment]
@personalizedrefrigerator is there an easy way to let users test your MR? Also, if that's a generic problem caused by GTK3's inertia scrolling, do you think it's worth opening an issue in GTK?

Thanks!

**Comment by LittleHuba** [comment]
This might be solved by #3379

**Comment by LittleHuba** [comment]
Now that the pipelines for Windows are running again, we can trigger a build of this PR to get an installer with this fix. Give me a few minutes.

**Comment by PiiNGPooNG** [comment]
> 
> 
> Hello @PiiNGPooNG, thanks for the feedback! First I'd need to know if it's an issue in Xournal++ or in GTK3. Can you try with the GTK3 Demo and report whether it works? See [#3260 (comment)](https://github.com/xournalpp/xournalpp/issues/3260#issuecomment-890350546)

Dead link

**Comment by lb90** [comment]
> Dead link

True! Give me just some minutes...

**Comment by lb90** [comment]
Here we are: https://gitlab.gnome.org/lb90/gtk/-/jobs/1536818/artifacts/browse/_build/demos/gtk-demo/

**Comment by LittleHuba** [comment]
You can test the PR via this installer: https://artprodsu6weu.artifacts.visualstudio.com/A4bb312fe-95d7-456e-96fb-7669bb27c94b/ce85de88-4b71-44a0-8a57-2707c0b9a6e7/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL3hvdXJuYWxwcC9wcm9qZWN0SWQvY2U4NWRlODgtNGI3MS00NGEwLThhNTctMjcwN2MwYjlhNmU3L2J1aWxkSWQvMTU0MzYvYXJ0aWZhY3ROYW1lL3dpbmRvd3M1/content?format=file&subPath=%2Fxournalpp-setup.exe

Please be aware that you will have to enable the setting first. See the description of #3379 for instructions.

**Comment by PiiNGPooNG** [comment]
> 
> 
> Here we are: https://gitlab.gnome.org/lb90/gtk/-/jobs/1536818/artifacts/browse/_build/demos/gtk-demo/

No problems in the demo

**Comment by PiiNGPooNG** [comment]
> 
> 
> You can test the PR via this installer: https://artprodsu6weu.artifacts.visualstudio.com/A4bb312fe-95d7-456e-96fb-7669bb27c94b/ce85de88-4b71-44a0-8a57-2707c0b9a6e7/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL3hvdXJuYWxwcC9wcm9qZWN0SWQvY2U4NWRlODgtNGI3MS00NGEwLThhNTctMjcwN2MwYjlhNmU3L2J1aWxkSWQvMTU0MzYvYXJ0aWZhY3ROYW1lL3dpbmRvd3M1/content?format=file&subPath=%2Fxournalpp-setup.exe
> 
> Please be aware that you will have to enable the setting first. See the description of #3379 for instructions.

Using that option "fixes" the issue. Works for me as a temporary fix until inertial scrolling is fixed.

**Comment by PiiNGPooNG** [comment]
Just noticed something in a document that has enough chapters for the contents sidebar to become scrollable: In there touch scrolling with inertia seems to have no problems.

**Comment by LittleHuba** [comment]
So you are saying whether scrolling with inertia works is dependent on the size of the document? Quite interesting...

**Comment by PiiNGPooNG** [comment]
I guess I didn't explain it very well.

In here have no problems with scrolling using touch:

![image](https://user-images.githubusercontent.com/8688059/134891354-089925ae-38f1-43aa-a7fe-06872ea18b40.png)

Only in the main view does the scrolling bug around like in the video I linked in my first comment.

**Comment by LittleHuba** [comment]
Sorry, closed the issue by accident :sweat_smile: 

Okay, so it is something explicit to this one view. This is more in accordance with my expectations. Could you send me a screenshot of the input classes you assigned to the different input devices in `Edit->Preferences->Input System`?

**Comment by PiiNGPooNG** [comment]
![image](https://user-images.githubusercontent.com/8688059/134893872-6de3a10a-d340-493a-8417-2834de2f4926.png)

**Comment by laundmo** [comment]
I am experiencing the same issue, frequent jumps especialyl when zoomed in quite far. As reported above, using a second finger does solve it.

**Comment by henrybrink** [comment]
I can confirm this behavoir on a Surface Pro 7 with Windows 11 and the newest stable build. It doesn't happen with shorter touches but does happen as soon the touchscreen is touched a bit longer.

**Comment by toelke** [comment]
Disabling inertial scrolling fixes it for me.

**Comment by henrybrink** [comment]
> Disabling inertial scrolling fixes it for me.

Where did you find this setting? I'm not seeing it anywhere.

**Comment by toelke** [comment]
![image](https://user-images.githubusercontent.com/183276/154477977-5cb6ed04-a52a-4deb-944f-e5d49694149e.png)

**Comment by henrybrink** [comment]
> ![image](https://user-images.githubusercontent.com/183276/154477977-5cb6ed04-a52a-4deb-944f-e5d49694149e.png)

Thank you. I hadn't upgraded to the current nightly, now I can see the setting.
I can confirm the scrolling issue is fixed when I turn on this setting.

**Comment by LJPadilla** [comment]
I was having the same issue and disabling inertial scroll seemed to work for me as well.

**Comment by SysCrasher** [comment]
OS: Windows 11 Home
Device: Surface Pro 9

I had the same issue with latest stable Version (1.2.1).
The same behaviour during scrolling with holding my finger shows the issue.
Scrolling like this with my pen was no problem.
Disabling GTK solved it for me to.

**Comment by rzippo** [comment]
I just encountered this issue as soon as I tried this app on my Surface Pro 9, and the fix works for me as well.

My question is: why is this not disabled by default, at least on Windows builds?
